### PR TITLE
Upgrade demoshop to beta2 version

### PR DESCRIPTION
**Pull request by MattCzerner**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Keep demoshop upgraded with newest version of Framework
|New feature| No
|BC breaks| No
|Fixes issues| 
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: docker/php-fpm/Dockerfile
Comment (Miroslav-Stopka) [inline docker/php-fpm/Dockerfile:68]: Question, what about docker/php-fpm/production/Dockerfile?

Will be the application still functional in production if you dont modify Dockerfile for production?

File: composer.json
Comment (Miroslav-Stopka) [inline composer.json:37]: Wait, why composer.json is modified in this commit?

File: composer.json
Comment (Miroslav-Stopka) [inline composer.json:70]: I dont see any connection of this compose.json modification with the microservices??

File: app/config/parameters.yml.dist
Comment (Miroslav-Stopka) [inline app/config/parameters.yml.dist:19]: seems strange, but the original PR https://github.com/shopsys/shopsys/pull/494/files contains also the modifications of project-base/kubernetes/... Did you processed these modifications too?

File: composer.json
Comment (Miroslav-Stopka) [inline composer.json:103]: Just asking, was this modification necessary? I think it was. Am I right? If so, this modification should not be marked as optional in our UPGRADE.md. Actually, I'm probably against this mark optional in our UPGRADE.md. What do you think?

File: build-dev.xml
Comment (Miroslav-Stopka) [inline build-dev.xml:19]: Interesting, this `clean` target was not added as a dependency for `test-db-demo` while the domains phing targets modifications. Seems like PH does not updated UPGRADE.md in the past. But I think that you did a better job when you fixed it. +1

**Review by Miroslav-Stopka — CHANGES_REQUESTED** [review]
Hi @MattCzerner,
I have only a few questions, please check them.

Seems like this upgrade was easy thanks to detailed UPGRADE.md. What do you think?

File: docker/php-fpm/Dockerfile
Comment (MattCzerner) [reply docker/php-fpm/Dockerfile:68]: php.ini overrides is already added during build of production dockerfile

File: composer.json
Comment (MattCzerner) [reply composer.json:37]: Oh, yeah wrong commit. I wonder if these commits should be squashed all together at the end. If not I will rebase it before merging and fix this :)

File: composer.json
Comment (MattCzerner) [reply composer.json:70]: I answered you in previous comment :)

File: app/config/parameters.yml.dist
Comment (MattCzerner) [reply app/config/parameters.yml.dist:19]: No, because demoshop does not contains kubernetes manifests at this moment. This was discussed with @PetrHeinz that it will be added in the future.

File: composer.json
Comment (MattCzerner) [reply composer.json:103]: Well I dont know actually, the instruction in UPGRADE.md does not contain information about why should i do it. But i guess that if these version make application broken than it should not be optional for sure. Should I write some feedback to a person who wrote it into UPGRADE?

**Comment by MattCzerner** [comment]
Please review my comments and if it will be all right with you. We can discuss if we should merge all commits as one and if not i will perform rebase where I will make single commit for fork deletion.

File: composer.json
Comment (Miroslav-Stopka) [reply composer.json:103]: I suggest to stop using "optional" permanently, but this decission is not on me.

**Comment by Miroslav-Stopka** [comment]
CR is done. You should process the rebase right now and the final version should be tested.

**Review by Miroslav-Stopka — APPROVED** [review]

**Review by boris-brtan — CHANGES_REQUESTED** [review]
as for the last commit, there exists [this PR](https://github.com/shopsys/demoshop/pull/22/files) so there are also fixes for postgres version from 9.5 to 10 so please decide if you cherry pick these changes or resolve this PR first and then we do merge of this one because this last commit is incomplete
and also docker-sync for windows is not here but that change is highly optional but it is not absolutely in `UPGRADE.md`

**Comment by boris-brtan** [comment]
i would suggest to do it without tha change of -win and -mac confs and after that update CR TEST and MERGE @Miroslav-Stopka  PR

**Comment by MattCzerner** [comment]
Changes of @Miroslav-Stopka can be merged after this is finished. Actually it wont be even problem to rebase on it. 

Currently we got 2 not functional docker configurations in repository and we suggest to use them. I suggest to merge this PR as it is, make application functional and then solve the PR of #22

**Comment by boris-brtan** [comment]
This way is also OK, so not forget to mention this remaining thing.

**Review by boris-brtan — APPROVED** [review]

**Comment by boris-brtan** [comment]
TESTS:
production - OK
CI - OK
local development - OK

**Pull request merged**
